card_live_dashboard/layouts/figures.py

```diff
@@ -114,6 +122,257 @@ def totals_figure(data: CardLiveData, type_value: str, color_by_value: str) -> g
+        if num_sets == 0 or num_categories == 0:
+            fig = EMPTY_FIGURE
+        # or if there is way too much data to plot
+        elif num_categories > MAX_UPSET_CATEGORIES:
+            fig = go.Figure(layout={
+                                'xaxis': {'visible': False},
+                                'yaxis': {'visible': False},
+                                'annotations': [{
+                                'text': f"{num_categories} total {title} "
+                                         "categories with current selection. "
+                                         "<br>Please subset input data "
+                                         "further to display set "
+                                         "intersections.",
+                                'xref': 'paper',
+                                'yref': 'paper',
+                                'showarrow': False,
+                                'font': {'size': 16}
+                                }]
+                            })
+
+        else:
+			# filter to top MAX_UPSET_INTERSECTIONS sets by cardinality
+            if num_sets > MAX_UPSET_INTERSECTIONS:
+                truncated = True
+                upset_data.intersections = upset_data.intersections[:MAX_UPSET_INTERSECTIONS]
+            else:
+                truncated = False
+
+            # generate plot
+            fig = plotly_upset_plot(upset_data, title, truncated)
+            
+            # size for display as appropriate for the number of categories
+            fig.update_layout(height=get_figure_height(num_categories))
+
+    return fig
+
+
+def prepare_intersection_data(data: CardLiveData, type_value: str) -> upsetplot.UpSet:
+    """
+    Prepare the CardLiveData to generate intersection plots, specifcally
+    convert into an UpSet object containing all intersections and cardinalities
+    :param data: a CardLiveData object from which the rgi_parser is called
+    :param type_value: The category in RGI to plot set membersips for
+    :return: An upsetplot.UpSet class containing the intersections and category
+             memberships for creating a plotly based UpSet plot
+    """
+
+    totals_df = data.rgi_parser.get_column_values(data_type=type_value,
+                                                      values_name='categories',
+                                                      drop_duplicates=True)
+    totals_df = totals_df.dropna()
+    category_sets = totals_df.reset_index().groupby('filename')\
+                                   .agg(lambda x: tuple(x)).applymap(list)
+    category_sets = category_sets['categories']\
+                    .apply(lambda x: sorted(x)).sort_values().apply(tuple)
+    category_sets = category_sets.value_counts()
+
+    # convert to upset data
+    upset_data = upsetplot.from_memberships(category_sets.index,
+                                           category_sets.values)
+    upset_data = upsetplot.UpSet(upset_data,
```

Comment:
So with the issue I was getting, I was able to come up with how to reproduce it (when using the full dataset ~3000 samples):

* Set **RGI results cutoff** to *perfect*
* Select to view only **cellular organisms**
* Set the time period to **6 months**
* Set the RGI Intersections plot to **AMR gene**

![image](https://user-images.githubusercontent.com/200517/102639066-1b6ec280-411e-11eb-99f1-3df509e252ff.png)

Printing out the value of **upset_data** right before this line I get:

```
ROB-1
True    1
dtype: int64
```

So I wonder if it's failing when there's only one entry in `upset_data`?

Below is the full stack trace:

```
Traceback (most recent call last):
  File "/home/apetkau/workspace/card-live-dashboard/card_live_dashboard/callbacks/__init__.py", line 170, in update_all_figures
    main_pane_figures = build_main_pane(time_subsets[time_dropdown], organism_identification_method, fig_settings)
  File "/home/apetkau/workspace/card-live-dashboard/card_live_dashboard/callbacks/__init__.py", line 305, in build_main_pane
    fig_intersections = figures.rgi_intersection_figure(data,
  File "/home/apetkau/workspace/card-live-dashboard/card_live_dashboard/layouts/figures.py", line 143, in rgi_intersection_figure
    upset_data = prepare_intersection_data(data, type_value)
  File "/home/apetkau/workspace/card-live-dashboard/card_live_dashboard/layouts/figures.py", line 209, in prepare_intersection_data
    upset_data = upsetplot.UpSet(upset_data,
  File "/home/apetkau/miniconda3/envs/card-live-dashboard/lib/python3.8/site-packages/upsetplot/plotting.py", line 285, in __init__
    self.totals) = _process_data(data,
  File "/home/apetkau/miniconda3/envs/card-live-dashboard/lib/python3.8/site-packages/upsetplot/plotting.py", line 93, in _process_data
    df = _check_index(df)
  File "/home/apetkau/miniconda3/envs/card-live-dashboard/lib/python3.8/site-packages/upsetplot/plotting.py", line 73, in _check_index
    for level in df.index.levels):
AttributeError: 'Index' object has no attribute 'levels'
```

Reply:
Should be fixed.